docs/DeveloperGuide.md

```diff
@@ -270,58 +270,115 @@ _{Explain here how the data archiving feature will be implemented}_
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1. User enters a command to add an opening
+2. Ultron adds the opening to its list of openings
+   Use case ends.
 
-    Use case ends.
 
 **Extensions**
 
-* 2a. The list is empty.
+1a. Ultron detects an error in the entered data.
+    1a1. Ultron informs the user that the data is invalid
+    Use case ends
+
+**Use case:  UC 2 - Deleting an opening**
+
+**MSS**
+
+1. User enters a command to delete an opening
+2. Ultron deletes the opening from its list of openings
+   Use case ends.
+
+
+**Extensions**
+
+1a. Ultron detects an error in the entered data.
+    1a1. Ultron informs the user that the data is invalid
+    Use case ends
+
+**Use case:  UC 3 - Editing an opening**
+
+**MSS**
+
+1. User enters a command to edit an opening
+2. Ultron edits the opening based on the user input parameters
+   Use case ends.
+
+
+**Extensions**
+
+1a. Ultron detects an error in the entered data.
+    1a1. Ultron informs the user that the data is invalid
+    Use case ends
+
+**Use case:  UC 4 - See all openings**
+
+**MSS**
+
+1. User enters a command to list all openings
+2. Ultron shows the user a list of all the openings that the user has added
+   Use case ends.
+
+
+**Extensions**
+
+1a. List is empty
+    1a1. Ultron informs the user that the list is currently empty
+    Use case ends
+
+**Use case:  UC 5 - See all openings with a specific status**
+
+**MSS**
+
+1. User enters a command to list all openings of a specific status
+2. Ultron shows the user a list of all the openings that match the status given by the user
+   Use case ends.
 
-  Use case ends.
 
-* 3a. The given index is invalid.
 
-    * 3a1. AddressBook shows an error message.
+**Extensions**
+
+1a. No opening contains the status specified by the user
+    1a1. Ultron informs the user that no openings of that status exist
+    Use case ends
+
 
-      Use case resumes at step 2.
 
 *{More to be added}*
 
 ### Non-Functional Requirements
 
-1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
-2.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
-3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+1. Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
+2. The app should be fast and respond within 100ms when users input commands.
+3. The app should start up and load the list from a saved file within 1 second of opening.
 
 *{More to be added}*
 
 ### Glossary
 
 * **Mainstream OS**: Windows, Linux, Unix, OS-X
-* **Private contact detail**: A contact detail that is not meant to be shared with others
+* **Opening**: Item representing an internship opening
```

Comment:
Perhaps could change to "Item containing all relevant details for a particular internship application"